types/index.d.ts

```diff
@@ -45,6 +45,8 @@ declare namespace StyleDictionary {
   }
 
   interface Config {
+    parsers?: Parser[];
+    transform?: Record<string, Partial<Named<Transform>>>;
```

Comment:
I'm not 100% sure what the expected usage of this `transform` record is. Based on the `component-cti` example, I see only `transformer` property has been defined: https://github.com/amzn/style-dictionary/blob/main/examples/advanced/component-cti/config.js#L26 (as opposed to including all possible fields: `name`, `matcher`, type`, `transitive`)

Reply:
So.... this is a substitute for calling `.registerTransform()`. It needs to match the internal structure for a transform which I believe is `Transform` in the current index.d.ts file.

Reply:
https://github.com/amzn/style-dictionary/blob/main/examples/advanced/component-cti/config.js#L40-L46

Reply:
Right... so I'm still a bit unclear about this I'm afraid! 

`registerTransform` takes `Named<Transform>>` as it's input: https://github.com/amzn/style-dictionary/blob/3.0/types/index.d.ts#L170 (i.e. the Transform, plus a field called `name`). 

However, `transform` in `Config` appears to take an object that maps the names of the transforms to the transform definitions. E.g.:

```ts
transform: {
  'attribute/cti': CTITransform
},
```

I typed this as `Record<string, Partial<Named<Transform>>>`. It's kind of equivalent to this which already exists:

```ts
interface Transforms {
  [name: string]: Transform;
}
```

So I've changed this to use `Transforms`, which I think is better. The only gotcha I think will be that the CTI example will have a type error, because only the `transformer` field is supplied, it should really be supplying `matcher` and `type` as well. _If_ however, it's possible to provide only some of those fields, then we should probably reflect this with `Partial` as in the original example (with or without `Named` as required). Sorry that's probably all rather confusing! 

Take a look at the new version and let me know what you think.

Reply:
If this is not helpful, I apologize in advance, as I'm doing like 7 things at once and have about 15 minutes left here before I have to be on childcare duty. I didn't want to leave you hanging until tomorrow though.... so hopefully this helps.

You have already seen registerTransform and registerTransformGroup, but for some additional examples, there are some great ones here (specifically in build.js): https://github.com/amzn/style-dictionary/tree/main/examples/advanced/custom-transforms

Now what does this have to do with the transform property in the config? Great question. When registerTransform is called, it simply does this: https://github.com/amzn/style-dictionary/blob/main/lib/register/transform.js#L59-L63

Wait, but that still isn't config - except, config works very similarly:
- https://github.com/amzn/style-dictionary/blob/main/bin/style-dictionary#L84
- https://github.com/amzn/style-dictionary/blob/main/lib/extend.js#L59-L138

That last link is a bit hard to interpret, but it is just modifying the StyleDictionary object and extends it by adding the configuration key/value pairs as properties (without overwriting).

Thus the type for transform should be the same in both places... I think. My amateur typescript skills showing again.

I hope this helps. If not, let me know and I can try to take a deeper look later.

Reply:
Hopping in here. Taking a look at the 'component-cti' example, @yoniholmes is correct and the example is wrong (sorry!). The example should have at least a `type` attribute, `matcher` on a Transform is optional. If `matcher` is omitted, it will match all tokens. It looks like the what you have currently looks good to me. The `transform` attribute in the config takes named Transforms, and a Transform is the same whether used on there or in `.registerTransform` the only difference is where the `name` of the Transform goes. In `.registerTransform` it is added in a `name` attribute, when applied directly to the config the `name` is the key in the `transform object. That is a long-winded way of saying I think this looks good and the 'component-cti' example is wrong and should be failing.

Reply:
Thanks @dbanksdesign I wasn't sure about the example. Appreciate the clarity.